### DISCUSSION TITLE
Dynamic import the integration

### DISCUSSION THREAD

**Discussion by kk-growthbox** [verbb/formie#1087] [Ideas]
Hi,
Currently we use the intergration of Google reCaptcha in our formie form. However the form is hiding in a modal initially. So if there is a way that we can dynamically load the recaptcha when it is needed?

**Comment by engram-design** [comment]
This issue is, it's more than just the captcha's you have to worry about, but other fields that require JS that might not be initialised if the content is within a modal.

Are you using a JS framework of any kinda? Vue and React will probably differ a little depending on the virtual DOM. It depends if the `formie.js` script is being run on page load, or if you're lazy-loading the script along with the HTML. reCaptcha should be smart enough so that it only initializes when the captcha placeholder is visible on the page.

As such, there's not an event you can call to initialize _just_ reCaptcha. But you can call an event to initialize the form's JS.

```js
const $form = document.querySelector(".my-selector form");
this.FormieInstance.initForm($form);
```

Which you would call when the modal is shown.

But as I said, this largely depends on how you're showing the modal, if Vue/React/etc are involved, lazy-loading and more. Also refer to the [docs](https://verbb.io/craft-plugins/formie/docs/developers/javascript-api) on JS options available to you.
